### PR TITLE
Update OWNERS labels for cluster-lifecycle and scheduling

### PR DESCRIPTION
**Pull request by timothysc**
**What this PR does / why we need it**:
Updates auto labeling to make everyone's lives easier. 

**Special notes for your reviewer**:

**Release note**:

```
NONE
```

/cc @kubernetes/sig-cluster-lifecycle-pr-reviews @kubernetes/sig-scheduling-pr-reviews

### CODE REVIEW

File: cmd/kubeadm/OWNERS [hunk 1]
```diff
@@ -12,5 +12,7 @@ reviewers:
 - jamiehannaford
 - kad
 - xiangpengzhao
-- mattmoyer
```
Comment (kad) [inline cmd/kubeadm/OWNERS]: Removing Matt is intentional or accidental ?

Comment (timothysc) [reply cmd/kubeadm/OWNERS]: he asked yes

**Pull request merged**